httpclient5-testing/src/test/java/org/apache/hc/client5/testing/extension/sync/UnixDomainProxyServer.java

```diff
@@ -0,0 +1,136 @@
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ * ====================================================================
+ *
+ * This software consists of voluntary contributions made by many
+ * individuals on behalf of the Apache Software Foundation.  For more
+ * information on the Apache Software Foundation, please see
+ * <http://www.apache.org/>.
+ *
+ */
+
+package org.apache.hc.client5.testing.extension.sync;
+
+import org.newsclub.net.unix.AFUNIXServerSocket;
+import org.newsclub.net.unix.AFUNIXSocket;
+
+import java.io.File;
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.net.Socket;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.CountDownLatch;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.concurrent.CompletableFuture.supplyAsync;
+
+public final class UnixDomainProxyServer {
+    private final int port;
+    private final ExecutorService executorService;
+    private final Path socketPath;
+    private final CountDownLatch serverReady = new CountDownLatch(1);
+    private volatile AFUNIXServerSocket serverSocket;
+
+    public UnixDomainProxyServer(final int port) {
+        this.port = port;
+        this.executorService = Executors.newCachedThreadPool();
+        this.socketPath = (new File("proxy.sock")).toPath();
+    }
+
+    public void start() {
+        executorService.submit(this::runUdsProxy);
+        try {
+            serverReady.await();
+        } catch (final InterruptedException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    public Path getSocketPath() {
+        return socketPath;
+    }
+
+    public void close() {
+        try {
+            serverSocket.close();
+            executorService.shutdownNow();
+            if (!executorService.awaitTermination(5, TimeUnit.SECONDS)) {
+                throw new RuntimeException("Failed to shut down");
+            }
+        } catch (final InterruptedException | IOException ex) {
+            throw new RuntimeException(ex);
+        }
+    }
+
+    private void runUdsProxy() {
+        try {
+            Files.deleteIfExists(socketPath);
+        } catch (final IOException ignore) {
+        }
+
+        try {
+            try (final AFUNIXServerSocket server = AFUNIXServerSocket.bindOn(socketPath, true)) {
```

Comment:
Is socket path deleted when the socket is closed?

Reply:
Yes, you should delete it. I looked into the same recently in Go. Common recommendation: https://dev.to/douglasmakey/understanding-unix-domain-sockets-in-golang-32n8.
Same in unbound: https://github.com/NLnetLabs/unbound/blob/81f3de4da2e4a35d04465471b436f8253719796f/services/listen_dnsport.c#L980-L995

Reply:
OK, done